### ISSUE TITLE
DPoP support in keycloak.js

### ISSUE THREAD

**Issue by mposolda** [keycloak/keycloak#33945]

**Comment by VinodAnandan** [comment]
@mposolda DPoP example implementation  = https://github.com/AxaFrance/oidc-client/

Also, related: We have Web Cryptography API with [non-extractable keys](https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/generateKey#extractable)

Thanks to @dteleguin

**Comment by smandlavdiya** [comment]
It is very much expected feature in the adapter and as @VinodAnandan has shared references are good starting point.

**Comment by michele6000** [comment]
HI all, i have a problem with DPoP: 
2024-12-21 23:35:58,295 ERROR [org.keycloak.services.error.KeycloakErrorHandler] (executor-thread-1) Uncaught server error: java.lang.NullPointerException: Cannot invoke "org.keycloak.protocol.oidc.OIDCAdvancedConfigWrapper.isUseDPoP()" because "clientConfig" is null

When configuring a client without dpop, and using it from keycloak-client official (java)


my keycloak.json is:

{
  "realm": "master",
  "auth-server-url" : "XXXXXXXXXX",
  "resource" : "authz-client",
  "credentials": {
    "secret": "XXXXXXXXXXX"
  }
}

client in keycloak:
<img width="360" alt="image" src="https://github.com/user-attachments/assets/f0d7fad4-d772-4a0d-a1b7-e3fd7ee732d2" />

**Comment by smandlavdiya** [comment]
> HI all, i have a problem with DPoP: 2024-12-21 23:35:58,295 ERROR [org.keycloak.services.error.KeycloakErrorHandler] (executor-thread-1) Uncaught server error: java.lang.NullPointerException: Cannot invoke "org.keycloak.protocol.oidc.OIDCAdvancedConfigWrapper.isUseDPoP()" because "clientConfig" is null
> 
> When configuring a client without dpop, and using it from keycloak-client official (java)
> 
> my keycloak.json is:
> 
> { "realm": "master", "auth-server-url" : "XXXXXXXXXX", "resource" : "authz-client", "credentials": { "secret": "XXXXXXXXXXX" } }
> 
> client in keycloak: <img alt="image" width="360" src="https://private-user-images.githubusercontent.com/37514630/397956398-f0d7fad4-d772-4a0d-a1b7-e3fd7ee732d2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ4NDcwNjgsIm5iZiI6MTczNDg0Njc2OCwicGF0aCI6Ii8zNzUxNDYzMC8zOTc5NTYzOTgtZjBkN2ZhZDQtZDc3Mi00YTBkLWExYjctZTNmZDdlZTczMmQyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjIyVDA1NTI0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI2YjlkZWU4YTM1OTAyOWE0NTFjNGU0ZTZkOWE2MWExYWRjYmJiM2I5Mjk2MmZhNWIzNzFiZjI2NjAxZGQ4ZjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.rVC-j6PkOyF8ptcQrhORYEn3fmEQqXcrYkg7L3PgfT0">

Yes it is not working based on client configuration instead I have to disable dpop feature itself , whereas it should have only enabled for the client configured for it.

**Comment by jonkoops** [comment]
This is not the place to discuss issues around DPoP, please keep things on topic and use the forums for questions.

**Comment by michele6000** [comment]
@jonkoops could you please link me the right issue if there is already one ?

**Comment by jonkoops** [comment]
I do not know of any such reported issue, file one or start a discussion.
